src/ir/context.rs

```diff
@@ -103,6 +103,10 @@ pub struct BindgenContext<'ctx> {
     /// item ids during parsing.
     types: HashMap<TypeKey, ItemId>,
 
+    /// Maps from a cursor to the item id of the named template type parameter
+    /// for that cursor.
+    named_types: HashMap<clang::Cursor, ItemId>,
```

Comment:
Seems like a good follow up. We also need to test whether it has any observable effect.